### PR TITLE
Move system message listeners to static methods

### PR DESCRIPTION
**Pull request by vitormattos**
related: https://github.com/nextcloud/spreed/issues/7152

### CODE REVIEW

File: lib/Chat/SystemMessage/Listener.php [hunk 1]
```diff
@@ -77,281 +77,303 @@ public function __construct(IRequest $request,
...
+			$listener->sendSystemMessage($room, 'lobby_non_moderators');
+		}
+	}
 
-		$dispatcher->addListener(Room::EVENT_AFTER_USERS_ADD, static function (AddParticipantsEvent $event) {
-			$room = $event->getRoom();
-			if ($room->getType() === Room::TYPE_ONE_TO_ONE) {
-				return;
-			}
+	public static function addXJoinedTheConversationWeneverUserId(AddParticipantsEvent $event): void {
```
Comment (nickvergessen) [inline lib/Chat/SystemMessage/Listener.php:254]: Typo:
```suggestion
	public static function addXJoinedTheConversationWheneverUserId(AddParticipantsEvent $event): void {
```
But actually lets go simple:
```suggestion
	public static function addSystemMessageUserAdded(AddParticipantsEvent $event): void {
```

File: lib/Share/RoomShareProvider.php [hunk 2]
```diff
@@ -155,7 +155,7 @@ public function create(IShare $share): IShare {
 		foreach ($existingShares as $existingShare) {
 			if ($existingShare->getSharedWith() === $share->getSharedWith()) {
 				// FIXME Should be moved away from GenericEvent as soon as OCP\Share20\IManager did move too
-				$this->dispatcher->dispatch(self::EVENT_SHARE_FILE_AGAIN, new GenericEvent($existingShare));
+				$this->dispatcher->dispatch(new GenericEvent($existingShare), self::EVENT_SHARE_FILE_AGAIN);
```
Comment (vitormattos) [inline lib/Share/RoomShareProvider.php:158]: @nickvergessen I think this solved a bug.

Comment (nickvergessen) [reply lib/Share/RoomShareProvider.php:158]: https://github.com/nextcloud/server/blob/215aef3cbdc1963be1bb6bca5218ee0a4b7f1665/lib/private/EventDispatcher/EventDispatcher.php#L86-L87

I think it explodes :P

Comment (vitormattos) [reply lib/Share/RoomShareProvider.php]: Crazy.
Psalm point to this file:
https://github.com/symfony/event-dispatcher-contracts/blob/main/EventDispatcherInterface.php#L32

I changed now the property to `OCP\EventDispatcher\IEventDispatcher` and solved the psalm issue.

**Review by nickvergessen — APPROVED** [review]

**Pull request merged**